ATM/ATM.py

```diff
@@ -0,0 +1,164 @@
+import time
+
+def set_pin():
+    user_input = None
+    while not pin_input_well_formed_check(user_input):
+        user_input = input("Please choose your four digit PIN. ")
+    return user_input
+
+
+def pin_input_well_formed_check(pin_input_to_be_checked):
+    if pin_input_to_be_checked == None:
+        return False
+    try:
+        int(pin_input_to_be_checked)
+    except:
+        print("You entered characters other than digits. Please try again.")
+        return False
+    if len(pin_input_to_be_checked) < 4:
+        print("You did not enter enough digits. Please try again.")
+        return False
+    if len(pin_input_to_be_checked) > 4:
+        print("You entered too many digits. Please try again.")
+        return False
+    if int(pin_input_to_be_checked) < 0:
+        print("Please refrain from trying to enter a negative number. Thank you.")
+        return False
+    else:
+        print("Thank you. You have set your pin to:" + pin_input_to_be_checked)
+        return True
+
+
+def pin_check(pin):
+    number_of_wrong_pin_entries = 0
+    while number_of_wrong_pin_entries < 3:
+        pin_entry = input("Please enter your four digit PIN ")
+        if pin_entry == pin:
+            return True
+        else:
+            number_of_wrong_pin_entries += 1
+            print("Number of wrong pin entries " + str(number_of_wrong_pin_entries))
+    print("You have entered a wrong PIN three times. Your card has been blocked. Have a nice day.")
+    quit()
+
+
+def set_initial_balance():
+    balance = input("Please enter your initial balance. ")
+    while not is_input_a_positive_float_with_a_max_of_two_decimal_places_check(balance):
+        balance = input("Please enter your initial balance. ")
+    print("Thank you.")
+    time.sleep(1)
```

Comment:
why this sleep here?

Reply:
Just for user experience. I thought it would feel nicer for the user if it takes a moment for the menu to appear.